firedrake/assemble.py

```diff
@@ -108,6 +108,48 @@ def assemble(expr, *args, **kwargs):
         raise TypeError(f"Unable to assemble: {expr}")
 
 
+class FormAssemblerBase(abc.ABC):
```

Comment:
Rename to `AbstractFormAssembler` to avoid confusion with `BaseFormAssembler`

Reply:
Fixed.